data/transactions/logic/ledger_test.go

```diff
@@ -829,3 +835,49 @@ func (l *Ledger) round() basics.Round {
 	}
 	return l.rnd
 }
+
+func (l *Ledger) LookupApplication(rnd basics.Round, addr basics.Address, aidx basics.AppIndex) (ledgercore.AppResource, error) {
```

Comment:
nice to have: comment that specifies which interface (or interfaces) this is aiming to implement.

e.g.

```go
// LookupApplication implements interface LedgerForCowBase in ledger/internal
```
Other similar comments might be applicable below.

Reply:
I don't dig this, but I suppose it's a larger discussion.  There's the Java way of thinking about interfaces, where there exists a public interface, and various implementations code to it (and state that with an `implements` clause). The Go way is that _clients_ create interfaces, often locally, to declare the things they need from other objects.  I think comments like this lead us toward the Java way of thinking that causes a proliferation of weird interfaces that are nearly the same.